parsing/unit_info.mli

```diff
@@ -124,10 +124,17 @@ val annot: t -> Artifact.t
     extension of its filename.
     Those functions purposefully do not cover all artifact kinds because we want
     to track which artifacts are assumed to be bundled together. *)
-val companion_cmi: Artifact.t -> Artifact.t
 val companion_obj: Artifact.t -> Artifact.t
 val companion_cmt: Artifact.t -> Artifact.t
 
+val companion_cmi: Artifact.t -> Artifact.t
+(** Beware that [companion_cmi a] strips all extensions from the
+ filename of [a] before adding the [".cmi"] suffix contrarily to
+ the other functions which only remove the rightmost extension.
+ In other words, the companion cmi of a file [something.d.cmo] is
+ [something.cmi] and not [something.d.cmi].
+ This is done for bacward compatibility reason for pack users.
```

Comment:
```suggestion
 This is done for backward compatibility reason for pack users.
```
Although I don't agree about the reason; it is because cmi files with extra dots cannot be looked up by the type-checker, so we know that the cmi companion has to be of this form. This only impacts packs because `companion_cmi` is only used for packs.

Note that if you do `ocamlc -c -o foo.d.cmo foo.ml` and there's no `foo.mli`, the compiler produces a useless `foo.d.cmi`. No need to fix this here (this happens with previous versions too), but the fact that we have three different functions in `Unit_info` to find the name of a cmi file is probably a bad sign.